src/librustc_session/output.rs

```diff
@@ -44,58 +44,6 @@ fn is_writeable(p: &Path) -> bool {
     }
 }
 
-pub fn find_crate_name(sess: Option<&Session>, attrs: &[ast::Attribute], input: &Input) -> String {
```

Comment:
Hilariously this would be fine if you kept it where it was when you moved this file.

Reply:
@eddyb. I think I tried to do that, but it depends on stuff in rustc, and rustc depends on rustc_session, so it caused a circular dependency.

Reply:
What stuff from `rustc`, specifically?

Reply:
Oops sorry. It uses rustc_metadata and rustc_ast, which I believe both transistively depend on rust_session through other crates.

Reply:
It uses `rustc_metadata` for a thing that I think should probably be in `rustc_session` (https://github.com/rust-lang/rust/pull/69965#discussion_r392677735), and I don't think `rustc_ast` should depend (transitively) on `rustc_session`, but rather the other way around.

If I'm wrong that means some stuff was moved out of `rustc::session` when it was moved to `rustc_session`, so PM me and we can sort it out then.